### PR TITLE
implemented methods for ODEs to solve in terms of bessel and Airy's  …

### PR DESCRIPTION
**Pull request by RituRajSingh878**
Currently, sympy does not support the ODEs solution in term of Bessel function.
So for this, I have taken help from pr #10870 and Implemented  ODEs in term of Bessel and Airy's function.

```
Eq(x ** 2*Derivative(y(x),x,x)-(ax+b)*y(x))
The above equation now has solution c1*Ai(x)+x2*Bi(x)
where Ai(x) and Bi(x) are Airy's functions of first and second kind respectively

```
```
Eq(x**2*f(x).diff(x,2)+x*f(x).diff(x)+(x**2-a**2)*f(x),0)
The above equation now has solution 
                 ⎛   ____   ⎞             ⎛   ____   ⎞
                 ⎜  ╱  2    ⎟             ⎜  ╱  2    ⎟
f(x) = C₁⋅besselj⎝╲╱  a  , x⎠ + C₂⋅bessely⎝╲╱  a  , x⎠


where besselj(n,x) and bessely(n,x) are Bessel's functions of first and second kind respectively
```

#### Release Notes

- solvers
  - implemented methods for ODEs to solve in terms of bessel and Airy's function.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Can you base this on the commits from #10870?

You can squash them to a single commit but make sure it still records the same author. Then commit your changes on top.

File: sympy/solvers/tests/test_ode.py [hunk 1]
```diff
@@ -3414,3 +3414,25 @@ def test_issue_15913():
...
+    C1, C2 = symbols("C1 C2")
+    from sympy.functions import besselj, bessely
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - 4)*f(x)
+    assert dsolve(eq) == Eq(f(x), C1*besselj(2, x) + C2*bessely(2, x))
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 +25)*f(x)
+    assert classify_ode(eq) == ('2nd_linear_bessel', '2nd_power_series_regular')
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2)*f(x)
+    assert dsolve(eq) == Eq(f(x), C1*besselj(0, x) + C2*bessely(0, x))
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (81*x**2 -S(1)/9)*f(x)
+    assert dsolve(eq) == Eq(f(x), C1*besselj(1/3, 9*x) + C2*bessely(1/3, 9*x))
```
Comment (supreet11agrawal) [inline sympy/solvers/tests/test_ode.py]: Perhaps, you should Simpify these terms. Eg. `S(1)/3`.
Python2.7 sometimes gives an error if not simpified

**Review by supreet11agrawal — CHANGES_REQUESTED** [review]

**Comment by RituRajSingh878** [comment]
> Can you base this on the commits from #10870?
> 
> You can squash them to a single commit but make sure it still records the same author. Then commit your changes on top.

Sure

**Comment by RituRajSingh878** [comment]
> Can you base this on the commits from #10870?
> 
> You can squash them to a single commit but make sure it still records the same author. Then commit your changes on top.

okkk

File: sympy/solvers/ode.py [hunk 2]
```diff
@@ -307,8 +307,9 @@
     "nth_linear_constant_coeff_variation_of_parameters",
     "nth_linear_euler_eq_nonhomogeneous_variation_of_parameters",
     "Liouville",
-    "order_reducible",
```
Comment (oscarbenjamin) [inline sympy/solvers/ode.py]: Looks like you've accidentally removed the order_reducible method...

File: sympy/solvers/ode.py [hunk 3]
```diff
@@ -638,7 +639,6 @@ def recur_len(l):
         hints = _desolve(eq, func=func,
             hint=hint, simplify=True, xi=xi, eta=eta, type='ode', ics=ics,
             x0=x0, n=n, **kwargs)
-
```
Comment (oscarbenjamin) [inline sympy/solvers/ode.py]: Please restore this blank line

File: sympy/solvers/ode.py [hunk 4]
```diff
@@ -1351,25 +1351,30 @@ class in it.  Note that a hint may do this anyway if
...
-        if r:
-            matching_hints['order_reducible'] = r
-
-        # Any ODE that can be solved with a combination of algebra and
-        # integrals e.g.:
-        # d^3/dx^3(x y) = F(x)
-        r = _nth_algebraic_match(reduced_eq, func)
-        if r['solutions']:
-            matching_hints['nth_algebraic'] = r
-            matching_hints['nth_algebraic_Integral'] = r
```
Comment (oscarbenjamin) [inline sympy/solvers/ode.py]: Here also you've removed order_reducible accidentally.

**Comment by oscarbenjamin** [comment]
I think that the way you based this PR on #10870 wasn't quite right.

I think it is probably better to do this on a new branch (needing a new PR).

1. Add the original authors repo as a remote (`git remote add`) and fetch their branches (`git fetch`)
2. Checkout the branch `git checkout remotename/branchname` where this work was done. Then checkout a branch (`git checkout -b newname`) from there. Push that branch up to your fork (`git push -u origin newname`).
3. At this point you have your own branch with their commits in so you can delete the remote you added (`git remote rm`).
4. Now apply your work onto the code, commit and push.
5. Open a new PR. Make a comment here referring to the new PR and vice-versa.
6. Close this PR.

As you become more proficient some of these steps could be streamlined with `git diff` and `git apply`.

**Comment by RituRajSingh878** [comment]
I have made another pr #16581 .

**Pull request closed without merging**